### ISSUE TITLE
ftplugin/man.vim gives prompt when redirecting input

### ISSUE THREAD

**Issue by cmcaine** [neovim/neovim#5474]
- `nvim --version`:
`````
NVIM 0.1.5
Build type: RelWithDebInfo
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.1.5/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
`````
- Vim (version: ) behaves differently?
Yes, man.vim distributed with vim 8 works as expected.
- Operating system/version:
Arch
- Terminal name/version: `rxvt-unicode-256color`

### Actual behaviour

I'm trying to use nvim as a man page reader: `MANPAGER='bash -c "nvim -c \"set ft=man\" -u NORC </dev/tty <(col -bx)"' man vim`

This is an unusual construction to avoid the remarks nvim makes when reading from stdin when you use a more conventional piped expression:
````
$ MANPAGER='sh -c "col -bx | nvim -c \"set ft=man\" -"' man vim
Nvim: Reading from stdin...
"-stdin-" 304L, 17185C"-stdin-" 304L, 17185C
````
With the `</dev/tty <(col -bx)` construction, nvim clears the screen and outputs these lines at the bottom of the screen:
````
"/dev/fd/63" [fifo/socket] 304L, 17185C
"man://vim(1)" [Not edited] line 1 of 304 --0%-- col 1
Press ENTER or type command to continue
````
Pressing enter gives the man page as expected.

### Expected behaviour

nvim should go straight to the man page without a Press ENTER prompt.

With vim 8 (using the default vimrc to source man.vim): `MANPAGER='bash -c "vim -c \"set ft=man\" </dev/tty <(col -bx)"' man vim`. This version gives no "Press ENTER" prompt and goes straight to the man page.

After `sudo swap /usr/share/nvim/runtime/ftplugin/man.vim /usr/share/vim/vim80/ftplugin/man.vim`, nvim gives the expected result and vim the bad result. `swap` just swaps two files [[source]](https://gist.github.com/cmcaine/1b5fab9e9e958877166742f42d5b831d).

### Steps to reproduce using `nvim -u NORC`

`MANPAGER='bash -c "nvim -c \"set ft=man\" -u NORC </dev/tty <(col -bx)"' man nvim`

**Comment by justinmk** [comment]
> After sudo swap /usr/share/nvim/runtime/ftplugin/man.vim /usr/share/vim/vim80/ftplugin/man.vim

So the difference is somewhere in the `ftplugin`? I ask because man.vim in nvim has an autoload and plugin script as well.

> to avoid the remarks nvim makes when reading from stdin ...
> Nvim: Reading from stdin...

FWIW I've long wanted to eliminate that message. We could instead start the UI and load buffer...

**Comment by cmcaine** [comment]
I saw that autoload and plugin were there too, but I didn't modify or move them at all when I swapped the `ftplugin`. So I don't think they're the problem.

The "reading from stdin" message is pretty much useless. I'm in favour of removing it too - you can hardly read from stdin by accident.

**Comment by cmcaine** [comment]
How can I help resolve this issue?

I could try building nvim without the warnings and submit a PR, but not sure how contentious that will be. If there's a reasonable chance that it'll be accepted, I'm happy to do that.

I could try to work out what the old and new man.vims are doing differently, but seems like maintainer would be able to do it easier. If @nhooyr doesn't have the time right now, I could give it a go.

**Comment by justinmk** [comment]
Preferably the latter

**Comment by nhooyr** [comment]
I reinstalled macOS recently so I have no dev environment. I'm resetting all my dotfiles and rebuilding them. Too much spaghetti :(

You can go ahead and do it yourself if you want, I don't mind reviewing the PR.

**Comment by cmcaine** [comment]
See #5509

It is sufficient to add `silent` to the `execute 'file man://'.ref` call in `ftplugin/man.vim`. The versions of man.vim in master `NVIM v0.1.6-265-g7505300` are not backwards compatible with my package manager's version of nvim `NVIM v0.1.5`, but the same fix works, if applied to the v0.1.5 versions of man.vim.
